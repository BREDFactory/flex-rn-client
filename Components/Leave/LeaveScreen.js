// @flow
import React from 'react';

import { View } from 'react-native';
import { AsyncStorage } from 'react-native';
import { Button, Card, Text } from 'react-native-elements';
import config from "../../config/api";
import server from "../../config/server";
import styles from "./LeaveScreenStyles";
import type { State, Props } from "./LeaveScreenType";

type Payload = {
  name: string,
  fname: string,
  id_user: string,
  id_place: string,
  historical: string | Array<object>
};

class LeaveScreen extends React.Component<Props, State> {
  static navigationOptions = {
    title: "Leave"
  };

  constructor() {
    super();
    this.state = {
      name: "",
      fname: "",
      id: "",
      place: "",
      debug: "",
      historical: ""
    };
  }

  leavePlace(ctx) {
    const { name, fname, id, place, historical } = ctx.state;
    ctx = ctx || window;

    const payload = {
      name,
      fname,
      id_user: id,
      id_place: place,
      historical,
    };
    fetch(server.address, {
      method: "POST",
      body: JSON.stringify(payload),
      headers: {
        "Content-Type": "application/json",
        "x-access-token": config.token
      }
    })
      .then(res => {
        if (res.ok) return res.json();
        else ctx.setState({ debug: "ERROR" });
      })
      .then(data => {
        ctx.state.debug = "";
        AsyncStorage.setItem("USER", JSON.stringify(ctx.state));
        ctx.goTo("Profile");
      });
  }

  componentDidMount() {
    AsyncStorage.getItem("USER", (err, result) => {
      if (err || result == null) this.goTo("Login");
      else {
        this.setState(JSON.parse(result));
        const userId: string = JSON.parse(result).id;
        fetch(`${server.address}users/${userId}`, {
          method: "GET",
          headers: {
            "Content-Type": "application/x-www-form-urlencoded",
            "x-access-token": config.token
          }
        })
          .then(res => res.json()) // transform data to json
          .then(data => {
            this.setState({ historical: data[0].historical });
          })
      }
    });
  }

  goTo(str: string) {
    const navigation = this.props.navigation;
    navigation.popToTop();
    navigation.navigate(str);
  }

  render() {
    const { fname, name, id, place } = this.state;
    return (
      <Card style={styles.view}>
        <Text>
          {fname}
          {name}[
{id}
          ]:
          {place}
        </Text>
        <Button
          style={styles.button}
          fontWeight="bold"
          borderRadius={15}
          backgroundColor="#5167A4"
          color="#fff"
          title="Leave place"
          onPress={() => this.leavePlace(this)}
        />
      </Card>
    );
  }
}
export default LeaveScreen
