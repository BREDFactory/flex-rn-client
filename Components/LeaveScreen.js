import React from 'react'

import { View, Text, Button } from 'react-native'
import { AsyncStorage } from 'react-native';
import config from '../config/api';
import server from '../config/server';

class LeaveScreen extends React.Component {
  static navigationOptions = {
    title: 'Leave',
  };

  constructor(){
    super();
    this.state = {
      name: '',
      fname: '',
      id:'',
      place:'',
      debug:'',
    }
  }

  goTo(str) {
    const navigation = this.props.navigation;
    navigation.popToTop();
    navigation.navigate(str);
  }

  leavePlace(ctx) {
    var ctx = ctx || window;

    var payload = {
        name: ctx.state.name,
        fname: ctx.state.fname,
        id_user: ctx.state.id,
        id_place: ctx.state.place
    };

    fetch(server.address , {
      method: "POST",
      body: JSON.stringify(payload),
      headers: {
        "Content-Type": "application/json",
        "x-access-token": config.token
      }
    })
    .then(function(res){
      if(res.ok)
        return res.json();
      else
        ctx.setState({debug:'ERROR'});
    })
    .then(function(data){
      ctx.state.debug = 'Leave';
      ctx.state.place = '';
      AsyncStorage.setItem('USER', JSON.stringify(ctx.state));
      ctx.goTo('Profile');
    });
  }

  componentDidMount(){
    AsyncStorage.getItem('USER', (err, result) => {
      if(err || result == null)
        this.goTo('Login');
      else {
        this.setState(JSON.parse(result));
      }
    });
  }

  render() {
    return (
      <View style={{justifyContent:'center', alignItems:'center'}}>
        <Text>{this.state.fname} {this.state.name} [{this.state.id}]: {this.state.place}</Text>
        <Button title='Leave place'
        color="#000"
        onPress={() => this.leavePlace(this)}/>
      </View>
    )
  }
}
export default LeaveScreen;
