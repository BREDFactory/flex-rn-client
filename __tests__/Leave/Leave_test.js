import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import LeaveScreen from "../../Components/Leave/LeaveScreen";

const navigation = { navigate: jest.fn(), popToTop: jest.fn() };

it("renders correctly", () => {
  const tree = renderer
    .create(<LeaveScreen navigation={navigation} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
