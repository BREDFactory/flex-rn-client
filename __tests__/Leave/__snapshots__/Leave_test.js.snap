// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      
      
      [
      
      ]:
      
    </Text>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          undefined,
          Object {
            "borderRadius": 15,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#5167A4",
              },
              Object {
                "borderRadius": 15,
              },
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  Object {
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Leave place
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
